schematic/schemas/create_json_schema.py

```diff
@@ -0,0 +1,895 @@
+    def _update_nodes_to_process(self, nodes: list[str]) -> None:
+        """Updates the nodes to process with the input nodes
+
+        Arguments:
+            nodes: Nodes to add
+        """
+        self._nodes_to_process += nodes
+
+
+def create_json_schema(  # pylint: disable=too-many-arguments
+    dmge: DataModelGraphExplorer,
+    datatype: str,
+    schema_name: str,
+    write_schema: bool = True,
+    schema_path: Union[str, None] = None,
+    jsonld_path: Optional[str] = None,
+    use_property_display_names: bool = True,
+    use_valid_value_display_names: bool = True,
+) -> dict[str, Any]:
+    """
+    Creates a JSONSchema dict for the datatype in the data model.
+
+    This uses the input graph starting at the node that corresponds to the input datatype.
+    Starting at the given node it will(recursively):
+    1. Find all the nodes this node depends on
+    2. Find all the allowable metadata values / nodes that can be assigned to a particular
+        node (if such a constraint is specified on the schema).
+
+    Using the above data it will:
+    - Cerate properties for each attribute of the datatype.
+    - Create properties for attributes that are conditionally
+        dependent on the datatypes attributes
+    - Create conditional dependencies linking attributes to their dependencies
+
+    Arguments:
+        dmge: A DataModelGraphExplorer with the data model loaded
+        datatype: the datatype to create the schema for.
+            Its node is where we can start recursive dependency traversal
+            (as mentioned above).
+        schema_name: Name assigned to JSON-LD schema (to uniquely identify it via URI
+            when it is hosted on the Internet).
+        write_schema: whether or not to write the schema as a json file
+        schema_path: Where to save the JSON Schema file
+        jsonld_path: Used to name the file if the path isn't supplied
+        use_property_display_names: If True, the properties in the JSONSchema
+          will be written using node display names
+        use_valid_value_display_names: If True, the valid_values in the JSONSchema
+          will be written using node display names
+
+    Returns:
+        JSON Schema as a dictionary.
+    """
+    graph_state = GraphTraversalState(dmge, datatype)
+
+    json_schema = JSONSchema(
+        schema_id="http://example.com/" + schema_name,
+        title=schema_name,
+        description=dmge.get_node_comment(node_label=datatype),
+    )
+
+    while graph_state.are_nodes_remaining():
+        if not graph_state.is_current_node_processed():
+            _process_node(
+                json_schema=json_schema,
+                graph_state=graph_state,
+                use_property_display_names=use_property_display_names,
+                use_valid_value_display_names=use_valid_value_display_names,
+            )
+        graph_state.move_to_next_node()
+
+    logger.info("JSON schema successfully generated from schema.org schema!")
```

Comment:
The usage of the logger is light throughout this process. I would like to see more debug log messages around so we can more effectively watch the process as it is executing & when we need to debug what's going on we could get away with looking at the debug logs instead of needing to run the code locally in a debugger session.